scripts/zf_mkdist

```diff
@@ -84,13 +85,13 @@ build_and_copy_artifacts() {
     artifact_dir="$2"
     if [ "${build_type}" == "debug" ]; then
         echo "Building debug binaries"
-        make_args=
+        make_args="${build_opts}"
     elif [ "${build_type}" == "shim" ]; then
         echo "Building socket shim"
-        make_args="ZF_DEVEL=1 shim"
+        make_args="${build_opts} ZF_DEVEL=1 shim"
     else
         echo "Building ndebug binaries"
-        make_args=NDEBUG=1
+        make_args="${build_opts} NDEBUG=1"
```

Comment:
I imagine you've been very particular about positioning the user provided `build_opts` before those we want for a given build type, such that we overwrite whatever is provided in `build_opts`. Do you know if it's the case that `make` takes the value it sees last, e.g., so that `make A=1 A=3` would mean `make` only thinks `A=3`?

As an extension, I wonder if we should try to improve the fault-tolerance of these build types to explicitly exclude some things. For example, it might be good to define `NDEBUG=0` in "debug" to make sure these build targets can't be corrupted by bad user input.

Reply:
In your example, I think `A=3` is the "strongest" and even beats the environment variables, i.e. `A=2 make A=1 A=3` should still think `A=3`.

My motivation to expose this variable was to enable parallelism. I probably would have taken a different path, such as what `onload_build` does - it computes the `parallel` variable automatically, but I noticed an unused `build_opts` and gave it a go.

I'm not keen on adding a new `debug` target in this specific case because:

1. We would probably need to handle `ZF_DEVEL=1` for the `shim` target the same way, and there might be more configuration knobs added soon?
2. It would make it look less similar to the Onload build scripts, which allow for the export of variables in envs. I would keep the Onload and TCPDirect scripts somewhat similar!

I see the use of the `parallel` variable as a safer alternative. What do you reckon?

Reply:
> I see the use of the `parallel` variable as a safer alternative

Sounds good to me! It does sound like a tricky problem to handle properly otherwise.